unblob/sandbox.py

```diff
@@ -0,0 +1,118 @@
+import ctypes
+import sys
+import threading
+from pathlib import Path
+from typing import Callable, Iterable, Optional, Type, TypeVar
+
+from structlog import get_logger
+from unblob_native.sandbox import (
+    AccessFS,
+    SandboxError,
+    restrict_access,
+)
+
+if sys.version_info >= (3, 10):
+    from typing import ParamSpec
+else:
+    from typing_extensions import ParamSpec
+
+from unblob.processing import ExtractionConfig
+
+logger = get_logger()
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+class Sandbox:
+    """Configures restricted file-systems to run functions in.
+
+    When calling ``run()``, a separate thread will be configured with
+    minimum required file-system permissions. All subprocesses spawned
+    from that thread will honor the restrictions.
+    """
+
+    def __init__(
+        self,
+        config: ExtractionConfig,
+        log_path: Path,
+        report_file: Optional[Path],
+        extra_restrictions: Iterable[AccessFS] = (),
+    ):
+        self.restrictions = [
+            # Python, shared libraries, extractor binaries and so on
+            AccessFS.read("/"),
```

Comment:
Does it also imply execution of third party tools?

Reply:
Yes it does. We are using a high-level API, so we don't need to bother with that granularity.

https://docs.rs/landlock/0.4.1/src/landlock/fs.rs.html#106-115

In theory, we could filter execution, but it looks like a can of worms, as we need to be sure, what tools could be transitively executed

Reply:
> In theory, we could filter execution, but it looks like a can of worms, as we need to be sure, what tools could be transitively executed

Let's not go there. yet :)